unboundid/src/com/unboundid/ldap/sdk/ConnectThread.java

```diff
@@ -170,9 +173,7 @@ public void run()
       {
         try
         {
-          final int soTimeout =
-                  Math.max(0, (int) options.getResponseTimeoutMillis());
-          socket.setSoTimeout(soTimeout);
+          socket.setSoTimeout(SSL_HANDSHAKE_TIMEOUT);
```

Comment:
Why do we always want it to be set to 3000? Is that the recommended timeout that is always set by the controller when we were fetching it from options?

Reply:
It's not recommended setting but it's optimal. This fix is still experimental, not sure if it will fix the issue we see on production controller. If this works, I can make a bigger change and make it configurable. Right now limiting the change to just one file.